frontend/test/storybook/visual-regression/v-search-type-button.spec.ts

```diff
@@ -8,45 +8,59 @@ const pressedUrl = (showLabel: boolean) =>
   `${defaultUrl(showLabel)};pressed:true`
 
 const searchTypeButtonLocator = 'button[aria-haspopup="dialog"]'
+const getIcon = (page: Page) =>
+  page.locator('button[aria-haspopup="dialog"] svg').first()
 
 test.describe.configure({ mode: "parallel" })
 
+const goAndWaitForSvg = async (page: Page, url: string) => {
+  const response = page.waitForResponse(/\.svg/)
+  await page.goto(url)
+  await response
```

Comment:
Can this also be expressed (perhaps more simply) as:

```suggestion
  await Promise.all([
    page.waitForResponse(/\.svg/),
    page.goto(url),
  ])
```

It isn't immediately clear why `waitForResponse` needs to be called before but awaited after `page.goto`, even though I can sort of _guess_ why it is... but it seems abstract regardless. A comment here would help. Is it just that `waitForResponse` needs to be called before `page.goto` because it starts "capturing" responses until it is awaited, or something like that?

Reply:
This is the way that the Playwright docs suggest using `waitForResponse`. I went down the rabbit hole to find the reason for it.

Previously, the docs were using `Promise.all` as you suggest. The reasons for it are really well described in [this comment](https://github.com/playwright-community/eslint-plugin-playwright/issues/199#issuecomment-1950272527). However, they found at least 2 possible ways of incorrectly using the `waitForResponse`:
- changing the order of the calls (putting the `goto` before `waitForResponse`) would sometimes make it so that the response would come before the navigation, so this await would hang forever.
- users would put `await` before one of the calls inside the `Promise.all` call
This is why all docs [were rewritten](https://github.com/microsoft/playwright/pull/19154) to use the setup that I used in this PR.

@sarayourfriend, do you think it's better to use the idiomatic style used in Playwright docs, or replace it with `Promise.all`?

Reply:
> users would put await before one of the calls inside the Promise.all call

:sweat_smile: I wish that would just raise a type error but... I guess it can't. I suppose a linter could prevent this kind of error, though.

Let's go with the idiomatic style. If it's documented that way at the library level, that's more important.